### PR TITLE
Switch auto field stats to an item pipeline

### PR DESCRIPTION
**Pull request by Gallaecio**
I was asked to support non-standard, custom items for auto field stats.

To achieve that, it seemed easier to switch the implementation to an item pipeline.

It also feels like this feature would make sense in scrapy-poet now. But I am not 100% sure, and I also would not want to move it until scrapy-poet gained add-on support.

To do:
- [x] Merge and release https://github.com/zytedata/zyte-common-items/pull/102, and update `setup.py` and `tox.ini` accordingly.
- [ ] Wait for @asadurski to test it out in a real project.

### CODE REVIEW

File: scrapy_zyte_api/_poet_item_pipelines.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                    f"that item, and hence print the auto field stats that "
+                    f"you requested by enabling the ZYTE_API_AUTO_FIELD_STATS "
+                    f"setting. If {url_field!r} is the wrong URL field for "
+                    f"that item type, use the ZYTE_API_AUTO_FIELD_URL_FIELDS "
+                    f"setting to set a different field."
+                )
+            return
+
+        page_cls = self._registry.page_cls_for_item(url, cls)
+        cls = page_cls or cls
```
Comment (kmike) [inline scrapy_zyte_api/_poet_item_pipelines.py:66]: I find this line confusing: previously cls was an item class, now it's either a page class or an item class. Do you think it's possible to refactor it, to make the logic a bit more clear? A basic way could be to rename cls to item_cls before this line, and  only introduce `cls` at this line (`cls = page_cls or item_cls` at least makes it clear that cls is now). But maybe there are other approaches, e.g. involving some changes below.

File: scrapy_zyte_api/_poet_item_pipelines.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from scrapy.utils.misc import load_object
+from scrapy_poet import InjectionMiddleware
+from web_poet.fields import get_fields_dict
+from web_poet.utils import get_fq_class_name
+from zyte_common_items.fields import is_auto_field
+
+logger = getLogger(__name__)
+
+
+class ScrapyZyteAPIPoetItemPipeline:
```
Comment (kmike) [inline scrapy_zyte_api/_poet_item_pipelines.py:17]: Is the idea to bake all scrapy-zyte-api related features to this pipeline, if needed? Or is this pipeline for computing the stats for overridden fields? If it's the latter, should it be renamed, to reflect it?

**Comment by kmike** [comment]
The code looks good 👍 But I haven't checked how it works, and if it solves the issue :)

File: scrapy_zyte_api/_poet_item_pipelines.py (hunk 1 above)
Comment (Gallaecio) [reply scrapy_zyte_api/_poet_item_pipelines.py]: https://github.com/scrapy-plugins/scrapy-zyte-api/pull/216/commits/a89f3b72d3e6e28565292c11afed6c8d74c891ff ?

File: scrapy_zyte_api/_poet_item_pipelines.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+from scrapy.utils.misc import load_object
+from scrapy_poet import InjectionMiddleware
+from web_poet.fields import get_fields_dict
+from web_poet.utils import get_fq_class_name
+from zyte_common_items.fields import is_auto_field
+
+logger = getLogger(__name__)
+
+
+class ScrapyZyteAPIAutoFieldStatsItemPipeline:
```
Comment (kmike) [inline scrapy_zyte_api/_poet_item_pipelines.py:17]: Moving it to scrapy-poet might allow to make the name shorter :)

**Review by kmike — APPROVED** [review]

**Comment by Gallaecio** [comment]
Closing in favor of https://github.com/zytedata/zyte-common-items/pull/124.

**Pull request closed without merging**